airbnb_project/listings/harvester_app/harvester/spiders/airbnb_url_builder.py

```diff
@@ -0,0 +1,215 @@
+from datetime import datetime, timedelta
+
+"""
+Values for the months that are used in the url builder
+"""
+AirBnbMonths = [
+    "january",
+    "february",
+    "march",
+    "april"
+    "august",
+    "july",
+    "june",
+    "may",
+    "september",
+    "october",
+    "november",
+    "december",
+]
+
+
+class AirBnbURLBuilder:
+    """
+    A class for building AirBnb URLs for web scraping purposes.
+
+    This class provides methods to generate various types of AirBnb search URLs
+    for Vancouver, Canada, including flexible date searches and specific date ranges.
+    """
+
+    SCHEME = "https"
+    BASE_URL = "www.airbnb.ca"
+    HOMES_PATH = "s/Vancouver--Canada/homes"
+    DEFAULT_PARAMS = {
+        "ne_lat": {},
+        "ne_lng": {},
+        "sw_lat": {},
+        "sw_lng": {},
+        "zoom_level": {},
+        "zoom": {},
+        "search_by_map": "true",
+        "tab_id": "home_tab",
+        "refinement_paths[]": "/homes",
+        "query": "Vancouver, BC",
+        "place_id": "ChIJs0-pQ_FzhlQRi_OBm-qWkbs"
```

Comment:
This is google maps place id for Vancouver. For now it will be stay fixed, but in later stages of the project it should be changeable to other cities like Surrey or Burnaby.